MvvmCross/ViewModels/MvxNavigationViewModel.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MS-PL license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using MvvmCross.Logging;
+using MvvmCross.Navigation;
+
+namespace MvvmCross.ViewModels
+{
+    public abstract class MvxNavigationViewModel
+        : MvxViewModel
+    {
+        private IMvxLogProvider _logProvider;
```

Comment:
I think we should also provide IoC in here.

Reply:
`IoCProvider`? Mhmm I think we want to encourage users to use dependency injection in most cases 🤔

Reply:
Alright, but let's make `IMvxNavigationService` the first ctor parameter then, so that it is easier to refactor later if we ever decide to remove `IMvxLogProvider` from it.